docfx_yaml/writer.py

```diff
@@ -878,7 +878,11 @@ def visit_reference(self, node):
             if 'http' in node.attributes['refuri']:
                 self.add_text('[{}]({})'.format(node.astext(), node.attributes['refuri']))
             else:
-                self.add_text('@{}'.format(node.attributes['refuri'].replace('#', '')))
+                # no need '.html#id' ending in yml files
+                pos = node.attributes['refuri'].find('.html')
+                if pos != -1:
+                     node.attributes['refuri'] = node.attributes['refuri'][0: pos]
```

Comment:
no need. 'refuri' return string somelike:  [target].html#segment. so we need to remove ".html#segment" as template don't need them.